packages/dds/map/src/mapKernel.ts

```diff
@@ -353,6 +370,38 @@ export class MapKernel {
 		this.submitMapClearMessage(op, copy);
 	}
 
+	/**
+	 * Get the attribution about the target key
+	 * @param key - Key to track
+	 * @returns The Attribution Key if the attribution tracking is enabled and key exists, otherwise undefined
+	 * @alpha
+	 */
+	public getAttribution(key: string): AttributionKey | undefined {
+		return this.attribution?.get(key);
+	}
+
+	/**
+	 * Get the whole attribution table
+	 * @returns The attribution table if the attribution tracking is enabled, otherwise undefined
+	 * @alpha
+	 */
+
+	public getAllAttribution(): Map<string, AttributionKey> | undefined {
+		return this.attribution;
+	}
+
+	private setAttribution(message: ISequencedDocumentMessage, key: string): void {
+		if (this.options?.attribution?.track ?? false) {
```

Comment:
```suggestion
		if (this.options?.attribution?.track) {
```

undefined is already falsy